code/components/jomjol_fileserver_ota/server_file.cpp

```diff
@@ -547,7 +547,7 @@ static esp_err_t download_get_handler(httpd_req_t *req)
                 /* Get value of expected key from query string */
                 if (httpd_query_key_value(buf, "readonly", param, sizeof(param)) == ESP_OK) {
                     ESP_LOGI(TAG, "Found URL query parameter => readonly=%s", param);
-                    readonly = param && strcmp(param,"true")==0;
```

Comment:
@jomjol Not sure what your intention here was with the old check but it would now throw this error: `the address of 'param' will never be NULL [-Werror=address]`